service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java

```diff
@@ -30,4 +36,23 @@ public String getChildrenLabel(List<Element<Child>> children) {
     public void addPageShowToCaseDetails(CaseDetails caseDetails, List<Element<Child>> children) {
         caseDetails.getData().put("pageShow", children.size() <= 1 ? "No" : "Yes");
     }
+
+    public List<Element<Child>> updateFinalOrderIssued(List<Element<Child>> children,
+        String orderAppliesToAllChildren, ChildSelector childSelector) {
+        if (YES.getValue().equals(orderAppliesToAllChildren)) {
+            children.forEach(child -> child.getValue().setFinalOrderIssued(YES.getValue()));
+        } else {
+            List<Integer> selectedChildren = childSelector != null ? childSelector.getSelected() : new ArrayList<>();
+            IntStream.range(0, children.size())
+                .forEach(index -> {
+                    Child child = children.get(index).getValue();
+                    if (!selectedChildren.isEmpty() && selectedChildren.contains(index)) {
+                        child.setFinalOrderIssued(YES.getValue());
+                    } else if (StringUtils.isEmpty(child.getFinalOrderIssued())) {
+                        child.setFinalOrderIssued(NO.getValue());
+                    }
+                });
```

Comment:
Thoughts on just using a good old fashioned for loop instead of an int stream
```suggestion
            for (int i = 0; i < children.size(); i++) {
                Child child = children.get(i).getValue();
                if (!selectedChildren.isEmpty() && selectedChildren.contains(i)) {
                    child.setFinalOrderIssued(YES.getValue());
                } else if (StringUtils.isEmpty(child.getFinalOrderIssued())) {
                    child.setFinalOrderIssued(NO.getValue());
                }
            }
```

Reply:
Agreed, I originally I was using mapToObj(), hence the use of IntStream. But now with forEach, it can be dropped.